### PR TITLE
Don’t write empty kern lookups

### PR DESCRIPTION
**Pull request by khaledhosny**
A font might have only LTR or RTL kerning, so check that we have kern
rules for each lookup before writing it since feaLib does not seem to
like empty lookups.

### CODE REVIEW

File: Lib/ufo2ft/kernFeatureWriter.py [hunk 1]
```diff
@@ -87,33 +87,41 @@ def write(self, linesep="\n"):
...
-            self._addKerning(lines, self.rtlClassPairKerning, rtl=True,
+            self._addKerning(rtlKern, self.rtlClassPairKerning, rtl=True,
                              ignoreZero=True)
+        if rtlKern:
+            lines.append("lookup kern_rtl {")
+            lines.extend(rtlKern)
             lines.append("} kern_rtl;")
             lines.append("")
 
         lines.append("feature kern {")
```
Comment (jamesgk) [inline Lib/ufo2ft/kernFeatureWriter.py]: Right before this, we may want to check

```python
if not (ltrKern or rtlKern):
    # no kerning pairs, don't write empty feature
    return ""
```

I think it's save to remove the similar check above.

File: Lib/ufo2ft/kernFeatureWriter.py [hunk 2]
```diff
@@ -87,33 +87,41 @@ def write(self, linesep="\n"):
...
-        if self.ltrScripts or not self.rtlScripts:
+        if (self.ltrScripts or not self.rtlScripts) and ltrKern:
```
Comment (jamesgk) [inline Lib/ufo2ft/kernFeatureWriter.py]: I don't think we need the old check here, just `if ltrKern:`

Comment (khaledhosny) [inline Lib/ufo2ft/kernFeatureWriter.py]: I was wondering about this. The font I’m working on has both RTL and LTR scripts, but kerning for the LTR script only. Right now my I end up with:
```
feature kern {
    lookup kern_ltr;
script latn;
language dflt;
lookup kern_ltr;
} kern;
```
Not sure why LTR kerning lookup is present twice, once with the global language systems and once with the LTR ones.

Comment (jamesgk) [inline Lib/ufo2ft/kernFeatureWriter.py]: OK right, this doesn't do what I intended. My goal was to make `kern_ltr` the default for all script/language pairs not mentioned in the feature source, and then make sure that all LTR scripts have the `kern_ltr` lookup and not the `kern_rtl` lookup while all RTL scripts have the opposite. I'm not sure how to do this with feature syntax. As it is, I think every script/language pair gets `kern_ltr` (and LRT scripts get it twice as you say).

Comment (khaledhosny) [inline Lib/ufo2ft/kernFeatureWriter.py]: Then we will have to specify all scripts explicitly. May be something like this:
```
if ltrKern and not rtlScripts:
   # write ltr lookup with no languages systems
elif rtlKern and not ltrScripts:
   # write rtl lookup with no languages systems
else:
   if ltrKern:
      # write ltr lookup with its languages systems
   elif rtlKern:
      # write rtl lookup with its languages systems
```

Comment (jamesgk) [inline Lib/ufo2ft/kernFeatureWriter.py]: That looks about right. My one remaining question would be: when there are both LTR and RTL scripts, there won't be a ltr lookup for the DFLT/dflt language system, right? Is that OK?

Comment (khaledhosny) [inline Lib/ufo2ft/kernFeatureWriter.py]: AFAK DFLT is used only when no lookups for the current script/language is found, so it should be fine. cc @behdad.

Comment (jamesgk) [inline Lib/ufo2ft/kernFeatureWriter.py]: In that case, could we only write the lookups with their language systems, and not put any in the default language system?

Comment (khaledhosny) [inline Lib/ufo2ft/kernFeatureWriter.py]: I’m not very sure about this, waiting for @behdad here.

Comment (khaledhosny) [inline Lib/ufo2ft/kernFeatureWriter.py]: I think we can merge this PR and deal with this issue separately.

**Pull request merged**